File: lib/repositories/exam_period_assignment_repository.dart
import '../../core/models/response_data.dart';
import '../../core/network/api_client.dart';
import '../../models/exam/exam_period_assignment.dart';

/// Repository phân công kỳ thi — tương đương `ExamPeriodAssignmentRepository`.
class ExamPeriodAssignmentRepository {
  ExamPeriodAssignmentRepository(this._api);

  final ApiClient _api;
  static const _controller = 'ExamPeriodAssignment';

  /// Danh sách phân công của thí sinh hiện tại (JWT).
  Future<ResponseData<List<ExamPeriodAssignment>>> getMyAssignments() {
    return _api.get<List<ExamPeriodAssignment>>(
      _api.buildPath(_controller, 'get-my-assignments'),
      fromJson: (json) => (json as List<dynamic>)
          .map((e) => ExamPeriodAssignment.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: ADMIN/Example-API-Flutter/Example.UserService.WebApp/Repository/ExamPeriodAssignmentRepository.cs
using Example.API.Client.Clients.ExamPeriodAssignment;
using Example.Common.Models;
using Example.UserService.API.Models;
using Example.UserService.WebApp.Repository.IRepository;

namespace Example.UserService.WebApp.Repository
{
    /// <summary>
    /// Facade truy vấn phân công kỳ thi của thí sinh đang đăng nhập.
    /// Không xử lý quy tắc hiển thị hay thời gian thi — chỉ ủy quyền sang <see cref="IExamPeriodAssignmentClient"/>.
    /// </summary>
    public class ExamPeriodAssignmentRepository : IExamPeriodAssignmentRepository
    {
        // Client HTTP gọi endpoint phân công kỳ thi trên UserService.API.
        private readonly IExamPeriodAssignmentClient _client;

        // Tiêm client qua DI; API lọc phân công theo user từ JWT.
        public ExamPeriodAssignmentRepository(IExamPeriodAssignmentClient client) => _client = client;

        // Ủy quyền lấy danh sách phân công thuộc user hiện tại (API lọc theo JWT).
        public Task<ResponseData<List<ExamPeriodAssignmentModel>>> GetMyAssignmentsAsync(CancellationToken cancellationToken = default)
            => _client.GetMyAssignmentsAsync(cancellationToken);
    }
}
